Node/Managing Node-NPM versions.md

```diff
@@ -0,0 +1,127 @@
+You can create a `.nvmrc` file containing a Node version number. Running `use`, `install`, `which` etc. commands will use the version specified in the file if none was provided in the command line. When using `nvm` and installing global packages, those packages will only be global for that specific version. eg. You currently use Node 16 and install `polyglot-cli`, if you switch to v18, the package won't be available there until you install it.
+
+```bash
+#Install NVM by running script
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash
+
+#Install latest Node version
+nvm install node
+
+#Install specific version of Node by specifying version or use an alias
+nvm install <version>
+nvm install <alias>
+
+#Check Node version currently in use and available versions
+nvm ls
+
+#Set default version
+nvm alias default <version>
+
+#Switching between Node versions
+nvm use <version>
+nvm use default
+
+#Uninstall Node version
+nvm uninstall <version>
+```
+
+#### n
+
+[n](https://github.com/tj/n) is a version manager similar to `nvm` and a good alternative
+You can create a `.n-node-version`, `.node-version` or `.nvmrc` file containing a Node version number. Running `n auto` will look for version specified in one of those files in that order and switch to it.
+
+```bash
+#Install n by running script
+npm install -g n
+
+#Install latest Node version
+n latest
+
+#Install specific version of Node by specifying version or use an alias
+n <version>
+n lts
+
+#Execute to check Node version currently in use and available versions
+n
+
+#Uninstall Node version(this does not affect cached versions)
+n uninstall <version>
+
+#Uninstall cached Node version
+n rm <version>
+```
+
+#### Using homebrew
+
+For this method you need to have [homebrew](https://brew.sh/) installed on your Mac(which you probably do). There might be additional steps required when switching versions using `brew` as switching is not supported, so you might need to uninstall and unlink the old version, and install and link new one.
+
+```bash
+#Install Node
+brew install node
+
+#Install specific version of node
+brew install node@<version>
+
+#Update Node
+brew upgrade node
+
+#Uninstall Node
+brew uninstall node
+```
+
+#### Using direct binary install
+
+Considering previously mentioned options, there is no particular advantage why you would use binary installation. You can still opt in to do so if you wish.
+
+#### avn - Automatic Version Switching
+
+[avn](https://github.com/wbyoung/avn) is a package for automatic version switching of Node using one of your Node version manager package. Currently supported are `nvm`, `n` and `nodebrew`. When you `cd` into a directory containing `.node-version` file, `avn` will automatically detect the change and use version manager of your choice to switch to the specified version.
+
+```bash
+#Install the packages
+npm install -g avn avn-nvm avn-n
+avn setup
+```
+
```

Comment:
Given the [`.node-version` support](https://github.com/shadowspawn/node-version-usage) and avn support, I'm guessing the good recommendation would be to use `n`?

Reply:
I think that in this chapter, we should make a recommendation which tool to use. Base on all of the tools listed here, `n` seems like the best option to me.